zokrates_core/src/flatten/mod.rs

```diff
@@ -754,9 +756,11 @@ impl<'ast, T: Field> Flattener<'ast, T> {
         statements_flattened: &mut FlatStatements<T>,
         expression: BooleanExpression<'ast, T>,
     ) -> FlatExpression<T> {
-        // check the cache
-        if let Some(c) = self.condition_cache.get(&expression) {
-            return (*c).into();
+        if !expression.is_constant() {
```

Comment:
I feel this is not needed if we made sure we did not insert any constants into the map? The map access below would already return None